src/main/java/net/minecraftforge/registries/holdersets/CompositeHolderSet.java

```diff
@@ -161,4 +162,35 @@ public Iterator<Holder<T>> iterator()
     {
         return this.getList().iterator();
     }
+    
+    /**
+     * Maps the sub-holdersets of this composite such that,if the list contains more than one element,
+     * each element of the list will serialize as an object.
+     * Prevents crashes from trying to serialize non-homogenous lists to NBT.
+     * @return List of holdersets with homogenous serialization behavior.
+     * Returns a new List if size > 1, otherwise returns the composite's existing List.
+     */
+    public List<HolderSet<T>> homogenize()
```

Comment:
The core team requests that this method be changed to avoid the use of the `OrHolderSet`-wrapping strategy when possible, by checking first if the list of components is homogeneous, akin to @KnightMiner's [comment](https://github.com/MinecraftForge/MinecraftForge/pull/9048/files#r973666496).